### PR TITLE
Feature/user menus creation

### PR DESCRIPTION
**Pull request by lmvittori**
-Create user menu selection view endpoints
-Create User menus HR view
Pending:
-Form saving
-Styles

### CODE REVIEW

File: MSLaunches.Backend/MSLaunches.Api/Controllers/LunchController.cs [hunk 1]
```diff
@@ -76,7 +77,32 @@ public LunchController(ILunchService lunchService)
...
+                {
+                    Date = lunch.Date,
+                    MealId = lunch.MealId
+
+                };
+                listLunches.Add(lunchToCreate);
+            }
+
+            var result = await _lunchService.CreateLunchesAsync(listLunches);
+
```
Comment (mravinale) [inline MSLaunches.Backend/MSLaunches.Api/Controllers/LunchController.cs]: seems like we always add more lunches, is this correct? if this is the case,  we need to change the signature to AddBatches

File: MSLaunches.Backend/MSLaunches.Data/EF/DatabaseMSLaunches.cs [hunk 2]
```diff
@@ -18,6 +16,15 @@ public static class DatabaseMSLunches
         public static void Initialize(WebApiCoreLunchesContext dbContext)
         {
             dbContext.Database.EnsureCreated();
+            var id1 = Guid.NewGuid();
```
Comment (Dongata) [inline MSLaunches.Backend/MSLaunches.Data/EF/DatabaseMSLaunches.cs]: this was removed to be more dynamic on a previous pr, pls refer to that

File: MSLaunches.Backend/MSLaunches.Domain/Services/MealTypeService.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        public MealTypeService(WebApiCoreLunchesContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        #endregion
+
+        #region Public Methods
+
+        public async Task<MealType> GetByIdAsync(Guid mealTypeId)
```
Comment (Dongata) [inline MSLaunches.Backend/MSLaunches.Domain/Services/MealTypeService.cs]: add summary or <inheritdoc />

File: MSLaunches.Frontend/src/app/menu/menu.component.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { filter, finalize } from 'rxjs/operators';
+
+import { MenuService } from './menu.service';
+
+
+import { MealGrouped } from '@app/core/Models/meal-grouped.model';
+import { MealType } from '@app/core/Models/meal-type.model';
+import { Meal } from '@app/core/Models/meal.model';
+import * as moment from 'moment';
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.component.ts]: you're not using moment

File: MSLaunches.Frontend/src/app/menu/menu.component.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+import * as _ from 'lodash';
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.component.ts]: you're not using lodash

File: MSLaunches.Frontend/src/app/menu/menu.component.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { filter, finalize } from 'rxjs/operators';
+
+import { MenuService } from './menu.service';
+
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.component.ts]: pls do not leave 2 break lines

File: MSLaunches.Frontend/src/app/menu/menu.service.ts [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { map, mapTo, mergeMap } from 'rxjs/operators';
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.service.ts]: mergemap and mapto are not used

File: MSLaunches.Frontend/src/app/menu/menu.service.ts [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+import { HttpClient } from '@angular/common/http';
+import { of } from 'rxjs/observable/of';
+import { tryStatement } from 'babel-types';
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.service.ts]: tryStatement is not used

File: MSLaunches.Frontend/src/app/menu/menu.service.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+import { Meal } from '../core/Models/meal.model';
+import { MealType } from '../core/Models/meal-type.model';
+import * as moment from 'moment';
+import * as _ from 'lodash';
```
Comment (Dongata) [inline MSLaunches.Frontend/src/app/menu/menu.service.ts]: lodash is not used

File: MSLaunches.Frontend/src/app/menu/menu.component.ts (hunk 4 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.component.ts]: Removed!

File: MSLaunches.Backend/MSLaunches.Domain/Services/MealTypeService.cs (hunk 3 above)
Comment (lmvittori) [reply MSLaunches.Backend/MSLaunches.Domain/Services/MealTypeService.cs]: Done!

File: MSLaunches.Frontend/src/app/menu/menu.component.ts (hunk 5 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.component.ts]: Removed

File: MSLaunches.Frontend/src/app/menu/menu.component.ts (hunk 6 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.component.ts]: Cleared up.

File: MSLaunches.Frontend/src/app/menu/menu.service.ts (hunk 7 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.service.ts]: Removed

File: MSLaunches.Frontend/src/app/menu/menu.service.ts (hunk 8 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.service.ts]: Removed

File: MSLaunches.Frontend/src/app/menu/menu.service.ts (hunk 9 above)
Comment (lmvittori) [reply MSLaunches.Frontend/src/app/menu/menu.service.ts]: Removed

File: MSLaunches.Backend/MSLaunches.Api/Controllers/LunchController.cs (hunk 1 above)
Comment (lmvittori) [reply MSLaunches.Backend/MSLaunches.Api/Controllers/LunchController.cs]: My misteke including Saving signatures here, was out of scope on this commit. 
I'll add AddBatches on the next PR!

**Pull request merged**